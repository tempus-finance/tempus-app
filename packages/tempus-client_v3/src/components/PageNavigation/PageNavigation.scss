.tc__page-navigation {  
  position: relative;
  padding: 10px 20px;
  border: none;
}

.tc__page-navigation__background {
  position: absolute;
  border: none;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-image: url('./svg/left-main-corner-smaller.svg'), url('./svg/right-main-corner-smaller.svg');
  background-repeat: no-repeat;
  background-position: left center, right center;

  &::before {
    content: '';
    position: absolute;
    top: 1px;
    bottom: 1px;
    left: 34px;
    right: 34px;
    background-color: var(--pageContainerBackground);
  }
}