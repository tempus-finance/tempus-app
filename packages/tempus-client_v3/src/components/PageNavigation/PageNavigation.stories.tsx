import { ComponentStory } from '@storybook/react';
import { CSSProperties } from 'react';
import { BrowserRouter } from 'react-router-dom';
import PageNavigation from './PageNavigation';

export default {
  title: 'PageNavigation',
  component: PageNavigation,
  argTypes: {},
};

const style: CSSProperties = {
  background: 'rgba(0, 0, 0, 0.1)',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  width: '100%',
  padding: '20px',
};

const Template: ComponentStory<typeof PageNavigation> = args => (
  <BrowserRouter>
    <div style={style}>
      <PageNavigation {...args} />
    </div>
  </BrowserRouter>
);

export const Example1 = Template.bind({});
Example1.args = {
  navigationLinks: [
    { text: 'Markets', path: '/' },
    { text: 'Portfolio', path: '/portfolio' },
  ],
};

export const Example2 = Template.bind({});
Example2.args = {
  navigationLinks: [
    { text: 'Markets', path: '/' },
    { text: 'LP', path: '/lp' },
    { text: 'Staking', path: '/staking' },
    { text: 'Portfolio', path: '/portfolio' },
  ],
};
