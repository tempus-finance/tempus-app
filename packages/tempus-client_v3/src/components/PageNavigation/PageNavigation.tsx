import { CSSProperties, FC, memo } from 'react';
import { Tab, Tabs } from '../shared';

import './PageNavigation.scss';

export interface PageNavigationLink {
  text: string;
  path: string;
}

export interface PageNavigationProps {
  navigationLinks: PageNavigationLink[];
}

const buttonWidthInPx = 186;

const PageNavigation: FC<PageNavigationProps> = ({ navigationLinks }) => {
  const containerStyle: CSSProperties = {
    width: `${navigationLinks.length * buttonWidthInPx}px`,
  };

  return (
    <div className="tc__page-navigation" style={containerStyle}>
      <div className="tc__page-navigation__background" />
      <Tabs size="large">
        {navigationLinks.map((navigationLink: PageNavigationLink) => (
          <Tab key={navigationLink.text} label={navigationLink.text} href={navigationLink.path} />
        ))}
      </Tabs>
    </div>
  );
};

export default memo(PageNavigation);
