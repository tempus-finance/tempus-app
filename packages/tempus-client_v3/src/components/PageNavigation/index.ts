import PageNavigation, { PageNavigationLink } from './PageNavigation';

export type { PageNavigationLink };
export default PageNavigation;
