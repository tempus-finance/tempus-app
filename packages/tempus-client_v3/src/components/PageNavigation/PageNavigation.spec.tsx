import { fireEvent, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import PageNavigation, { PageNavigationProps } from './PageNavigation';

const defaultProps: PageNavigationProps = {
  navigationLinks: [
    { text: 'Link 1', path: '/' },
    { text: 'Link 2', path: '/link2' },
  ],
};

const subject = (props: PageNavigationProps & { location: string }) => {
  window.history.pushState({}, 'Link 1', props.location);

  return render(
    <BrowserRouter>
      <PageNavigation {...props} />
    </BrowserRouter>,
  );
};

describe('PageNavigation', () => {
  it('renders a page navigation component with first tab selected', () => {
    const { container, getByRole } = subject({ ...defaultProps, location: '/' });

    const actualPageNavigation = container.querySelector('.tc__page-navigation');

    const firstTab = getByRole('link', { name: /Link 1/ });
    const secondTab = getByRole('link', { name: /Link 2/ });

    expect(firstTab).not.toBeNull();
    expect(secondTab).not.toBeNull();

    expect(actualPageNavigation).toMatchSnapshot();
  });

  it('renders a page navigation component with second tab selected', () => {
    const { container, getByRole } = subject({ ...defaultProps, location: '/link2' });

    const actualPageNavigation = container.querySelector('.tc__page-navigation');

    const firstTab = getByRole('link', { name: /Link 1/ });
    const secondTab = getByRole('link', { name: /Link 2/ });

    expect(firstTab).not.toBeNull();
    expect(secondTab).not.toBeNull();

    expect(actualPageNavigation).toMatchSnapshot();
  });

  it('navigates to a path when the corresponding tab is clicked', () => {
    const { container, getByRole } = subject({ ...defaultProps, location: '/' });

    const firstTab = getByRole('link', { name: /Link 1/ });
    const secondTab = getByRole('link', { name: /Link 2/ });

    expect(firstTab).not.toBeNull();
    expect(secondTab).not.toBeNull();

    expect(window.location.pathname).toBe('/');

    fireEvent.click(secondTab);

    expect(window.location.pathname).toBe('/link2');

    const actualPageNavigation = container.querySelector('.tc__page-navigation');
    expect(actualPageNavigation).toMatchSnapshot();
  });
});
