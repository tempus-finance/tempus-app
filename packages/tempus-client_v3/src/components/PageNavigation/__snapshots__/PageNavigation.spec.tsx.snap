// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavigation navigates to a path when the corresponding tab is clicked 1`] = `
<div
  class="tc__page-navigation"
  style="width: 372px;"
>
  <div
    class="tc__page-navigation__background"
  />
  <div
    class="tc__tabs tc__tabs__tabs-page-navigation"
  >
    <a
      class="tc__link tc__tabs__tab"
      href="/"
    >
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Link 1
      </div>
    </a>
    <a
      class="tc__link tc__tabs__tab"
      href="/link2"
    >
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Link 2
      </div>
    </a>
    <div
      class="tc__tabs__indicator-container"
    >
      <div
        class="tc__tabs__indicator"
        style="width: 50%; left: 50%;"
      />
    </div>
  </div>
</div>
`;

exports[`PageNavigation renders a page navigation component with first tab selected 1`] = `
<div
  class="tc__page-navigation"
  style="width: 372px;"
>
  <div
    class="tc__page-navigation__background"
  />
  <div
    class="tc__tabs tc__tabs__tabs-page-navigation"
  >
    <a
      class="tc__link tc__tabs__tab"
      href="/"
    >
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Link 1
      </div>
    </a>
    <a
      class="tc__link tc__tabs__tab"
      href="/link2"
    >
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Link 2
      </div>
    </a>
    <div
      class="tc__tabs__indicator-container"
    >
      <div
        class="tc__tabs__indicator"
        style="width: 50%; left: 0%;"
      />
    </div>
  </div>
</div>
`;

exports[`PageNavigation renders a page navigation component with second tab selected 1`] = `
<div
  class="tc__page-navigation"
  style="width: 372px;"
>
  <div
    class="tc__page-navigation__background"
  />
  <div
    class="tc__tabs tc__tabs__tabs-page-navigation"
  >
    <a
      class="tc__link tc__tabs__tab"
      href="/"
    >
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Link 1
      </div>
    </a>
    <a
      class="tc__link tc__tabs__tab"
      href="/link2"
    >
      <div
        style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Link 2
      </div>
    </a>
    <div
      class="tc__tabs__indicator-container"
    >
      <div
        class="tc__tabs__indicator"
        style="width: 50%; left: 50%;"
      />
    </div>
  </div>
</div>
`;
